pandas/core/dtypes/missing.py

```diff
@@ -206,7 +207,18 @@ def _isna(obj):
     elif isinstance(obj, ABCSeries):
         result = _isna_array(obj._values)
         # box
-        result = obj._constructor(result, index=obj.index, name=obj.name, copy=False)
+        if isinstance(obj.dtype, ArrowDtype):
```

Comment:
I would usually recommend to avoid any additional nested if statements, it makes it extra verbose and harder to read.
If you look at the signature of the constructor you can choose to pass `dtype=None` if you want to let pandas handle the type internally.